src/main/java/org/quiltmc/loader/api/QuiltLoader.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.loader.api;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.loader.api.entrypoint.EntrypointContainer;
+import org.quiltmc.loader.impl.QuiltLoaderImpl;
+
+import net.fabricmc.loader.api.EntrypointException;
+import net.fabricmc.loader.api.LanguageAdapter;
+
+import net.fabricmc.api.EnvType;
+
+/**
+ * The public-facing FabricLoader instance.
+ *
+ * <p>To obtain a working instance, call {@link #getInstance()}.</p>
+ *
+ * @since 0.4.0
+ */
+public interface QuiltLoader {
+	/**
+	 * Returns the public-facing Quilt Loader instance.
+	 */
+	static QuiltLoader getInstance() {
```

Comment:
I would like to eliminate the `getInstance` function and make every other function accessible through quilt loader's singleton as a static function.

If a need to have a swappable impl is needed in the future, we can slap a semi-private api behind the static method calls

Reply:
Just to check: do you mean you want to move from:

```java
public interface QuiltLoader {
    boolean isModLoaded(String id);
}
```
to
```java
public final class QuiltLoader {
    public static boolean isModLoaded(String id) { return QuiltLoaderImpl.isModLoaded(id); }
}
```
?

Technically we can already make QuiltLoaderImpl swappable by making `QuiltLoader.getInstance` return a delegate impl of QuiltLoader and then changing what that delegate object points to, so we don't actually need to change the public api at all yet.

Reply:
Yeah that seems fine

Reply:
Ok, I quite like that idea - since it makes it much easier for mods to use directly.

Reply:
Ok, implemented.